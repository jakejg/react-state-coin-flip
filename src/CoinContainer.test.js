import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import CoinCointainer from './CoinContainer';

test('renders without crashing', () => {
  render(<CoinCointainer />);
});

test('matches snapshot', () => {
  const {asFragment} = render(<CoinCointainer />);
  expect(asFragment()).toMatchSnapshot();
});


test('should update heads or tails count correctly', () => {
  jest.spyOn(Math, "random")
  .mockReturnValueOnce(0.2)
  .mockReturnValueOnce(0.75);

  const { queryByText } = render(<CoinCointainer />);
  fireEvent.click(queryByText("Flip"));
  expect(queryByText("Out of", {exact: false})).toHaveTextContent("Out of 1 flips there have been 1 heads and 0 tails");
  fireEvent.click(queryByText("Flip"));
  expect(queryByText("Out of", {exact: false})).toHaveTextContent("Out of 2 flips there have been 1 heads and 1 tails")

})