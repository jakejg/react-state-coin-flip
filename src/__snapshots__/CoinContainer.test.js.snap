// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Flip a coin!
    </h1>
    <button>
      Flip
    </button>
    <p>
      Out of 0 flips there have been 0 heads and 0 tails
    </p>
  </div>
</DocumentFragment>
`;
